src/model.js

```diff
@@ -406,6 +411,10 @@ export default class MediaPlayerObject {
           return this.callService(e, 'unjoin', {
             entity_id: entity,
           }, platform);
+        case PLATFORM.HEOS:
+          return this.callService(e, 'unjoin', {
+            entity_id: (typeof entity === 'string') ? entity : entity[0],
```

Comment:
This indicated that the ungroup (ungroup all) button won't work as expected.
Might be a limitation in the heos implementation, seems like it doesn't allow passing multiple entity ids to the `unjoin` service, can you confirm this?

If this is the case it would be interesting to try by sending multiple `unjoin` calls in sequence if `entity` is an array of entities to see if the intergratin can handle that.

Reply:
Yeah, it is a limitation in the HA integration, not in pyheos. In fact it is not according to HEOS API, but it seems to work very well, so I guess that's why they did it this way. My thought here is that when ungrouping more than a single player, it ungroups the master which will effectively dissolve the whole group. It works very well with my four players at least.

I tried looping several unjoins, but that seemed to break the system and result in errors (processing previous command replies).

Reply:
Great, if you've tested it and it seems to work that's great.